iXBRLViewerPlugin/localviewer.py

```diff
@@ -47,11 +57,16 @@ def launchLocalViewer(cntlr, modelXbrl):
             outDir = os.path.dirname(_archiveFilenameParts[0]) # it's a zip or package
         else: 
             outDir = modelXbrl.modelDocument.filepathdir
-        out = modelXbrl.modelDocument.basename + VIEWER_SUFFIX
-        iv.save(os.path.join(outDir, out))
+        # for IXDS, outPath must be a directory name, suffix is applied in saving files
```

Comment:
I don't think we want this.  If I'm manually processing a document set (rather than launching from the viewer), and give it a directory for output, I don't want it to mangle the filenames at all because:

a) It'll output files with a `.ixbrlview` suffix which a browser won't recognise
b) The JS won't be able to find the other files in the document set, because it expects to find them at the original filenames.  This works in localviewer, because localviewer rewrites requests for `foo.htm` to `foo.htm.ixbrlview` internally.

For command line operation, we should probably allow a user-defined suffix, but we probably want to insert it before the existing file extension (or replace the existing file extension).

Reply:
My code looks quite different from the above.

Reply:
Agreed to:
* Change "viewer suffix" to "basename suffix" and have it inserted before the last extension
* Make it an option, defaulting to nothing.
* LocalViewer mode can still use `.ixbrlview`, but it'll now be inserted before the existing extension, not after.  The local viewer server will need updating to match.
* Prevent overwriting existing files unless a "force" option is specified
* LocalViewer mode can use force mode by default (although we might want to change the suffix that it uses to be even more unique, e.g. `.xbrlview-localviewer` )